src/main/java/uk/nhs/hee/trainee/details/service/RabbitPublishService.java

```diff
@@ -51,6 +61,41 @@ public class RabbitPublishService {
     this.rabbitTemplate = rabbitTemplate;
   }
 
+  /**
+   * Set up the Rabbit confirmation callback handler.
+   */
+  @PostConstruct
+  public void postConstruct() {
+    rabbitTemplate.setConfirmCallback((correlation, ack, reason) -> {
+
+      if (correlation == null) {
+        return;
+      }
+
+      handleRabbitAcknowledgement(ack, correlation);
+    });
+  }
+
+  /**
+   * Log successfully acked messages and raise sentry alerts for nacked messages.
+   *
+   * @param isAck           was the message acknowledged (success) or not (failure).
+   * @param correlationData message correlation details.
+   */
+  void handleRabbitAcknowledgement(boolean isAck, CorrelationData correlationData) {
+    final String id = correlationData.getId();
+    outstandingConfirms.remove(id);
+    if (!isAck) {
+      log.info("Rabbit message for programme membership id '{}' got nack-ed", id);
+      AmqpException e = new AmqpException("Rabbit message for programme membership id '"
```

Comment:
A more specific error might be useful e.g. [AmqpNackReceivedException](https://docs.spring.io/spring-amqp/api/org/springframework/amqp/rabbit/core/AmqpNackReceivedException.html) - but we don't have the message to-hand at this point.